utils/lister.cc

```diff
@@ -20,7 +20,11 @@ lister::lister(file f, dir_entry_types type, walker_type walker, filter_type fil
         , _show_hidden(do_show_hidden) {}
 
 future<> lister::visit(directory_entry de) {
-    return guarantee_type(std::move(de)).then([this] (directory_entry de) {
+    return refresh_type(std::move(de)).then([this] (directory_entry de) {
+        // The entry was removed between readdir and stat, just skip it
+        if (!de.type.has_value()) {
+            return make_ready_future<>();
+        }
```

Comment:
Isn't de.type disengaged when the syscall doesn't return the type?

Reply:
It is, but syscall doesn't return the type only when the file is missing, so it should be skipped.

Reply:
getdents():

"
       Currently,  only  some  filesystems (among them: Btrfs, ext2, ext3, and ext4) have full support for returning the file type in d_type.  All applications must properly handle a
       return of DT_UNKNOWN.

"

Reply:
Sorry, I wasn't specific enough in my answer.

If syscall doesn't return a type, the de.type is disengaged, that's true. In that case the lister::refresh_type() calls seastar::file_type() which engages the type with the result of stat() syscall or returns disengaged type if the file is missing (stat() returned ENOENT errno). Thus, this check only sees disengaged type in the latter case.